src/Server.zig

```diff
@@ -1089,50 +1096,58 @@ pub fn updateConfiguration(
-                    .{ zls_version, zls_version.major, zls_version.minor, zig_version },
-                );
-            } else {
-                server.showMessage(
-                    .Warning,
-                    "ZLS {} requires at least Zig {s} but got Zig {}. Update Zig to avoid unexpected behavior.",
-                    .{ zls_version, build_options.minimum_runtime_zig_version_string, zig_version },
-                );
-            }
-        },
+    check: {
+        if (!options.resolve) break :check;
+        if (!std.process.can_spawn) break :check;
+        if (server.status != .initialized) break :check;
+
+        // TODO there should a way to suppress this message
+        if (server.config.zig_exe_path == null) {
+            server.showMessage(.Warning, "zig executable could not be found", .{});
+        } else if (server.resolved_config.zig_lib_dir == null) {
+            server.showMessage(.Warning, "zig standard library directory could not be resolved", .{});
+        }
+    }
+
+    check: {
+        if (!options.resolve) break :check;
+        if (server.status != .initialized) break :check;
+
+        switch (server.resolved_config.build_runner_version) {
+            .resolved, .unresolved_dont_error => break :check,
+            .unresolved => {},
+        }
+
+        const zig_version = server.resolved_config.zig_runtime_version.?;
+        const zls_version = build_options.version;
+
+        const zig_version_is_tagged = zig_version.pre == null and zig_version.build == null;
+        const zls_version_is_tagged = zls_version.pre == null and zls_version.build == null;
+
+        if (zig_version_is_tagged) {
+            server.showMessage(
+                .Warning,
+                "ZLS {} does not support Zig {}. A ZLS {}.{} release should be used instead.",
+                .{ zls_version, zig_version, zig_version.major, zig_version.minor },
+            );
+        } else if (zls_version_is_tagged) {
+            server.showMessage(
+                .Warning,
+                "ZLS {} should be used with a Zig {}.{} release but found Zig {}.",
+                .{ zls_version, zls_version.major, zls_version.minor, zig_version },
+            );
+        } else {
+            server.showMessage(
+                .Warning,
+                "ZLS {} requires at least Zig {s} but got Zig {}. Update Zig to avoid unexpected behavior.",
+                .{ zls_version, build_options.minimum_runtime_zig_version_string, zig_version },
+            );
+        }
     }
 
     if (server.config.prefer_ast_check_as_child_process) {
         if (!std.process.can_spawn) {
-            log.info("'prefer_ast_check_as_child_process' is ignored because your OS can't spawn a child process", .{});
+            log.info("'prefer_ast_check_as_child_process' is ignored because the '{s}' operating system can't spawn a child processes", .{@tagName(zig_builtin.target.os.tag)});
```

Comment:
```suggestion
            log.info("'prefer_ast_check_as_child_process' is ignored because the '{s}' operating system can't spawn child processes", .{@tagName(zig_builtin.target.os.tag)});
```